proof/meta.go

```diff
@@ -73,3 +95,30 @@ func (m *MetaReveal) Decode(r io.Reader) error {
 	}
 	return stream.Decode(r)
 }
+
+// ValidateMetaReveal validates the meta reveal.
+func ValidateMetaReveal(r *MetaReveal) error {
+	// A meta reveal is allowed to be nil.
+	if r == nil {
+		return nil
+	}
+
+	switch r.Type {
+	case MetaOpaque:
+		// Valid type, fall through.
+
+	default:
+		return ErrInvalidMetaType
```

Comment:
I think we want to still allow the other type, so this way people can start to experiment with how they should be defined.

Reply:
Okay, I see the point about being able to add other types (without changing the proto _or_ the proof version).
But then shouldn't we begin with a 1:1 mapping of the RPC enum to the native type so we _don't_ need any `switch` statements in any place? So basically changing `MetaOpaque MetaType = 1` to `MetaOpaque MetaType = 0` so it matches `META_TYPE_OPAQUE = 0;`?

Reply:
> FYI, with the code as is, adding a new meta reveal type would involve bumping the proof.Proof version.

Why's that the case? No interpretation is actually needed. You just need to re-create the TLV serialization then hash. In reality, the type here is more of a client side thing.

Reply:
Yeah, he meant because of the marshal and un-marshal functions that returned an error for unknown meta types. The way I solved it now (direct cast, no `switch` statement) should allow users to use custom types.